amino.go

```diff
@@ -398,6 +419,35 @@ func (cdc *Codec) UnmarshalBinaryBare(bz []byte, ptr interface{}) error {
 		return err
 	}
 
+	if rv.CanAddr() {
+		addr := rv.Addr()
+		if addr.Type().Implements(binaryUnmarshalerType) {
+			bz2 := bz
```

Comment:
Is this necessary? Lines 437-438 won't actually modify `bz` from the caller's POV.